### PR TITLE
Make error logs dependent on DCPS::DCPS_debug_level

### PR DESCRIPTION
**Pull request by jwillemsen**
* dds/DCPS/RTPS/Spdp.cpp:

### CODE REVIEW

File: dds/DCPS/RTPS/Spdp.cpp [hunk 1]
```diff
@@ -4202,10 +4272,11 @@ void Spdp::SpdpTransport::thread_status_task(const DCPS::MonotonicTimePoint& /*n
     StatusMap running;
     StatusMap removed;
     if (!local_thread_status_manager_.sync_with_parent(
-          *global_thread_status_manager_, running, removed) &&
-        DCPS::DCPS_debug_level) {
-      ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: Spdp::SpdpTransport::thread_status_task: "
-        "syncing with global thread status manager failed\n"));
+          *global_thread_status_manager_, running, removed)) {
+      if (DCPS::DCPS_debug_level > 0) {
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Spdp.cpp:4276]: Why is this changing?  The change doesn't seem to have any effect.

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: dds/DCPS/RTPS/Spdp.cpp (hunk 1 above)
Comment (jwillemsen) [reply dds/DCPS/RTPS/Spdp.cpp:4276]: The return was done in the past only when debug level was set

Comment (mitza-oci) [reply dds/DCPS/RTPS/Spdp.cpp:4276]: Right, I didn't see that detail before.

**Review by mitza-oci — APPROVED** [review]

File: dds/DCPS/RTPS/Spdp.cpp (hunk 1 above)
Comment (jwillemsen) [reply dds/DCPS/RTPS/Spdp.cpp:4276]: This was tricky code, took me also some time to see the initial issue when checking the debug log level in this file

**Pull request merged**